app/models/jira_issue.rb

```diff
@@ -1,103 +1,103 @@
-      issue.post_deploy_check_status = extract_custom_select_field_from_jira_data(jira_data, 12202)
-      issue.deploy_type = extract_custom_multi_select_field_from_jira_data(jira_data, 12501)
-      issue.long_running_migration = extract_custom_multi_select_field_from_jira_data(jira_data, 10601)
-
-      if jira_data.assignee
-        issue.assignee = User.create_from_jira_data!(jira_data.assignee)
-      end
-
-      if jira_data.respond_to?(:parent)
-        issue.parent_issue = create_from_jira_data!(JIRA::Resource::IssueFactory.new(nil).build(jira_data.parent))
+      transaction do
+        JiraIssue.find_or_initialize_by(key: jira_data.key).tap do |issue|
+          issue.summary = jira_data.summary.truncate(1024)
+          issue.issue_type = jira_data.issuetype.name
+          issue.status = jira_data.fields.dig('status', 'name')
+          issue.post_deploy_check_status = extract_custom_select_field_from_jira_data(jira_data, 12202)
+          issue.deploy_type = extract_custom_multi_select_field_from_jira_data(jira_data, 12501)
+          issue.long_running_migration = extract_custom_multi_select_field_from_jira_data(jira_data, 10601)
+
+          if jira_data.assignee
+            issue.assignee = User.create_from_jira_data!(jira_data.assignee)
+          end
+
+          if (parent_data = jira_data.try(:parent))
+            issue.parent_issue = create_from_jira_data!(JIRA::Resource::IssueFactory.new(nil).build(parent_data))
+          end
+        end
       end
-
-      issue
     end
 
     private
 
     def extract_custom_select_field_from_jira_data(jira_data, field_number)
       field_name = "customfield_#{field_number}"
-      if jira_data.fields[field_name]
-        jira_data.fields[field_name]['value']
-      end
+      jira_data.fields.dig(field_name, 'value')
     end
 
     def extract_custom_date_field_from_jira_data(jira_data, field_number)
       field_name = "customfield_#{field_number}"
-      if jira_data.fields[field_name]
-        Date.parse(jira_data.fields[field_name])
+      if (date = jira_data.fields[field_name])
+        Date.parse(date)
       end
     end
 
     def extract_custom_multi_select_field_from_jira_data(jira_data, field_number)
       field_name = "customfield_#{field_number}"
-      if jira_data.fields[field_name] # rubocop:disable Style/SafeNavigation
-        jira_data.fields[field_name].collect do |value|
-          value['value']
-        end.join ', '
+      if (field_values = jira_data.fields[field_name])
+        field_values.map { |value| value['value'] }.join(', ')
       end
     end
   end
 
+  def commits_for_push(push)
+    commits.joins(:commits_and_pushes).where(commits_and_pushes: { push_id: push.id })
+  end
+
+  def long_running_migration?
+    long_running_migration == 'Yes'
+  end
+
+  # TODO: This should use Rails Serialization
+  def deploy_types
+    if deploy_type
+      deploy_type.split(',').map_compact(&:strip)
```

Comment:
Why `map_compact` instead of `map`? Can `strip` ever return `nil`?

Reply:
That `compact` is definitely unnecessary here. I went ahead and removed it.